StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/ViewControllers/PaymentMethodFormViewController.swift

```diff
@@ -25,20 +25,6 @@ class PaymentMethodFormViewController: UIViewController {
         // TODO Copied from AddPaymentMethodViewController but this seems wrong; we shouldn't have such divergent paths for link and instant debits. Where is the setDefaultBillingDetailsIfNecessary call, for example?
         if let linkEnabledElement = form as? LinkEnabledPaymentMethodElement {
             return linkEnabledElement.makePaymentOption(intent: intent)
-        } else if let instantDebitsFormElement = form as? InstantDebitsPaymentMethodElement {
-            // we use `.instantDebits` payment method locally to display a
-            // new button, but the actual payment method is `.link`, so
-            // here we change it for the intent confirmation process
-            let paymentMethodParams = STPPaymentMethodParams(type: .link)
-            let intentConfirmParams = IntentConfirmParams(
-                params: paymentMethodParams,
-                type: .stripe(.link)
```

Comment:
one "notable" exception is that this has changed where the type is no long `.stripe(.link)` though the `STPPaymentMethodParams(type: .link)` remains

hope that is OK

link to the new change: 

https://github.com/stripe/stripe-ios/pull/3800/files#diff-305ea2bdb8f8647eb2bfc2a7f506c6cecba3250c1884cbd970556c9e08379dceR68-R70

Reply:
Both old and new code do the same thing.

If you look at the initializer called in the old code path, it will call this with `paymentMethodType` equal to `link`
```
    convenience init(type: PaymentSheet.PaymentMethodType) {
        switch type {
        case .stripe(let paymentMethodType):
            let params = STPPaymentMethodParams(type: paymentMethodType)
            self.init(params: params, type: type)
        ...
```

This is the same as the new initializer code you added:
```
    convenience init(type: PaymentSheet.PaymentMethodType) {
        switch type {
        ...
        case .instantDebits:
            let params = STPPaymentMethodParams(type: .link)
            self.init(params: params, type: type)
```

Reply:
@yuki-stripe 

if I understood correctly, the `STPPaymentMethodParams` is fine (that was the "though the STPPaymentMethodParams(type: .link) remains" part)

but `type` changed from `.stripe(.link)` to `.instantDebits`?

Reply:
Ah sorry, I see what you mean now.  I looked at the users of `type` here and it looks fine. Thanks for flagging.